### ISSUE TITLE
Formatting in the variable spinner box is only applied after focus is removed

### ISSUE THREAD

**Issue by stephenmcc** [ihmcrobotics/simulation-construction-set-2#96]
Suppose variable q_x starts at 0.0. I click up then down a couple times. Then the variable is at 2.7755575615628914E-17, but what's shown is only 2.77555756. Once I click to another part of the UI, the rounding/formatting is applied and it shows  2.776E-17. But it'd be nice if that was applied after each click to the spinner arrow, otherwise it's confusing.

**Comment by SylvainBertrand** [comment]
Yup I understand. It'll take some work as I think we need to change the underlying behavior of the JavaFX spinner 😢 
The main issue is that when editing the spinner, we want the raw value to prevent rounding to trigger actual change to the YoVariable. When clicking the up/down arrows, for whatever reason, JavaFX requests focus for the spinner text field which then appears as if the user is editing. I'll see if there is another approach to that.
